### PR TITLE
sensors: as5600: added as5600

### PR DESCRIPTION
**Pull request by uLipe**
magnetic angle sensor driver.

Signed-off-by: Felipe Neves <felipe.neves@linaro.org>

### CODE REVIEW

**Comment by uLipe** [comment]
@MaureenHelm @teburd  fixed the CI, PTAL :)

File: samples/sensor/ams_as5600/README.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.. _ams_as5600:
+
+AMS AS5600 Angle position sensor sample
```
Comment (MaureenHelm) [inline samples/sensor/ams_as5600/README.rst:3]: We are trying to stop propagating driver-specific sensor samples. Will you please look at making this sample driver-agnostic, similar to `samples/sensor/accel_polling` or `samples/sensor/magn_polling`?

Comment (uLipe) [reply samples/sensor/ams_as5600/README.rst:3]: @MaureenHelm looking to these samples I see in this case, the best option here is to remove this sample right? The shell usage would offer the same value and will not introduce more pollution to the sensor samples folder.

Other point is to remove this sample, and create a polling angle sensor  sample in a separate PR.

What do you think?

Comment (uLipe) [reply samples/sensor/ams_as5600/README.rst:3]: @MaureenHelm , just following up

Comment (MaureenHelm) [reply samples/sensor/ams_as5600/README.rst:3]: Either option is fine

**Comment by uLipe** [comment]
Sample removed as we agreed, a new unified polled angle sensor sample will be added instead in other PR, @MaureenHelm

**Comment by uLipe** [comment]
@teburd PTAL :)

**Comment by uLipe** [comment]
@MaureenHelm @teburd any update here?

**Comment by uLipe** [comment]
@microbuilder PTAL

**Comment by microbuilder** [comment]
I couldn't get this to show up on the ST B-U585:

```bash
$ west build -p auto -b b_u585i_iot02a zephyr/samples/sensor/sensor_shell -- -DCONFIG_AMS_AS5600=y
$ west flash
```

I have this connected to I2C using 3.3V for power, and the SCL/SDA pins on the Mikroe Bus Click header, which I routes the pins to the Arduino SCL/SDA pins, but scanning the bus I get:

```
uart:~$ i2c scan i2c@40005800
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- -- 
10: 10 -- -- -- -- -- -- -- -- -- -- -- -- -- 1e -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- 56 -- -- -- -- -- -- 5d 5e 5f 
60: -- -- -- -- -- -- -- -- -- -- -- 6b -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
7 devices found on i2c@40005800
```

I was expecting it on `0x3C` (or 0x6C shifted left 1 bit), neither of which appears.

Nothing on the other I2C bus either:

```
uart:~$ i2c scan i2c@40005400 
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:             -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
0 devices found on i2c@40005400
```

This is the board I'm using: https://wiki.seeedstudio.com/Grove-12-bit-Magnetic-Rotary-Position-Sensor-AS5600/

I think these work with 3.3V as well, and don't require 5V???

**Comment by uLipe** [comment]
@microbuilder could you please re-test using both 5V and 3.3V voltages? I was able to discover them by just wiring it to a regular ST discovery board :)

File: drivers/sensor/ams_as5600/ams_as5600.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	.channel_get = as5600_get,
+};
+
+#define AS5600_INIT(n)						\
+	static struct as5600_dev_data as5600_data##n;		\
+	static const struct as5600_dev_cfg as5600_cfg##n = {\
+		.i2c_port = I2C_DT_SPEC_INST_GET(n)	\
+	};	\
+									\
+	DEVICE_DT_INST_DEFINE(n, as5600_initialize, NULL,	\
```
Comment (MaureenHelm) [inline drivers/sensor/ams_as5600/ams_as5600.c:94]: Please use `SENSOR_DEVICE_DT_INST_DEFINE`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI

**Review by MaureenHelm — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Pull request merged**